libs/template/unpatch/src/lib/tests/unpatch.directive.spec.ts

```diff
@@ -116,4 +116,41 @@ describe(RxUnpatch.name, () => {
       addEventListener.mockRestore();
     }
   });
+
+  it('should re-apply all except provided negated event listeners', () => {
+    // Arrange
+    const fixture = TestBed.createComponent(TestComponent);
+    fixture.componentInstance.unpatch = ['!mouseenter'];
+    const appRef = TestBed.inject(ApplicationRef);
+    const div = fixture.debugElement.query(By.css('div'));
+    const addEventListener = jest.spyOn(
+      div.nativeElement,
```

Comment:
one question, what happens when you put `['!mouseenter', 'mouseenter']` or `['!mouseenter', 'mousemove']`?

Reply:
Good question.

My first thought here is, why would a user specify this? What is the intention, what is expected?

Secondly, when looking at the implementation, whenever the user specifies a negated event by adding the `!` prefix, all events except the negated ones are unpatched. Other non-negated events specified are basically ignored.

```typescript
const negatedEvents = this.events
        .filter((event) => event.startsWith('!'))
        .map((event) => event.replace('!', ''));

const nextEvents = negatedEvents.length
        ? zonePatchedEvents.filter(
            (zonePatchedEvent) => !negatedEvents.includes(zonePatchedEvent)
          )
        : this.events;
```

Do you think it would be a good idea to add unit tests to proof this behaviour and prevent possible regression in the future?

Reply:
so negatedEvents always wins? We just need to put this information in the docs so no one is actually confused. You could also think about throwing an error in devMode when some weird input is specified

Reply:
ok, gotcha.

I've updated the documentation, I hope it's more clear now.

Regarding validating input, I agree with that, we should also throw an error in dev mode when non-existing events are specified.